old_timesheet_migration/timesheet_migration.py

```diff
@@ -0,0 +1,186 @@
+    '''Receive time and the day 
+        Returns the start and end in correct format.'''
+
+    time_hours = (time * 50) // 60
+    time_minutes = (time * 50) % 60
+    start = datetime.strptime(column + 'T00:00:00-0300', "%d-%m-%YT%H:%M:%S%z")
+    start = start - timedelta(days=1)
+    end = start + timedelta(hours=time_hours, minutes=time_minutes)
+    return (start, end)
+
+def get_time(time):
+    '''Returns a float that represents the time.'''
+
+    if type(time) is str:
+        try:
+            time = float(time.replace('ha', '').replace(',', '.'))
+        except:
+            time = 0
+    return time
+
+def create_time_entries(time_entries, clean_timesheet):
+    '''Returns a dataframe with all time entries in path file.'''
+
+    timesheet_collumns = list(clean_timesheet.iloc[:, 4:])
+    for key, row in clean_timesheet.iterrows():
+        member_acronym = row[0].lower()
+        no_time_projects = ('Feriado', 'Falta justificada')
+        if row_valid(row[1], row[2], row[3], no_time_projects):
+            names = fix_row(row[1], row[2], row[3], no_time_projects)
+            for column in timesheet_collumns:
+                project_name = names['project']
+                activity_name = names['activity']
+                client_name = names['client']
+                time = get_time(row[column])
+                if time > 0:
+                    start, end = calculate_start_end(time, column)
+                    time_entries = time_entries.append({'member_acronym': member_acronym,
+                                                        'project_name': project_name,
+                                                        'activity_name': activity_name,
+                                                        'client_name': client_name,
+                                                        'start': start, 'end': end}, ignore_index=True)
+    return time_entries
+
+def get_entity_id(entity, where, update=None):
+    return globals().get(entity).update_or_create(where, update).id
+
+def fill_ids(time_entries):
+    '''Receives a data frame with all time entries without ids
+        Returns a data frame with ids.'''
+
+    member_ids = {}
+    members_acronym = time_entries.member_acronym.unique()
+    for acronym in members_acronym:
+        email = {'email': acronym + "@certi.org.br"}
+        member_ids[acronym] = get_entity_id('Member', {'acronym': acronym}, email)
+
+    project_ids = {}
+    project_names = time_entries.project_name.unique()
+    for name in project_names:
+        project_ids[name] = get_entity_id('Project', {'name': name})
+
+    activity_ids = {}
+    activity_names = time_entries.activity_name.unique()
+    for name in activity_names:
+        activity_ids[name] = get_entity_id('Activity', {'name': name})
+    
+    client_ids = {}
+    client_names = time_entries.client_name.unique()
+    for name in client_names:
+        client_ids[name] = get_entity_id('Client', {'name': name})
```

Comment:
@yvesfracari and @ribeirojose I think we can do this in one for loop with something like this:

```python
member_ids=project_ids=activity_ids=client_ids={}
members_acronym = time_entries.member_acronym.unique()
project_names = time_entries.project_name.unique()
activity_names = time_entries.activity_name.unique()
client_names = time_entries.client_name.unique()
# you need to add None to the list so they have the same size (make a function to do this?)
zip_infos = zip(members_acronym, project_names, activity_names, client_names)
for acronym, project_name, activity_name, client_name in zip_infos:
    if acronym is not None:
        email = {'email': acronym + "@certi.org.br"}
        member_ids[acronym] = get_entity_id('Member', {'acronym': acronym}, email)
    if project_name is not None:
        project_ids[project_name] = get_entity_id('Project', {'name': project_name})
    if activity_name is not None:
        activity_ids[activity_name] = get_entity_id('Activity', {'name': activity_name})
    if client_name is not None:
        client_ids[client_name] = get_entity_id('Client', {'name': client_name})
```

Not sure if this is more complex hahah. What do you guys think?